### ISSUE TITLE
Make some migration rules less specific

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/ph-scale#260]
While adding migration rules for https://github.com/phetsims/ph-scale/issues/258, @samreid, @zepumph and I experiemented with making some migration rules less specific, which seems to improve readability of the rules file.

Links for testing with previous versions:
* https://phet-io.colorado.edu/sims/ph-scale/1.5/
* https://phet-io.colorado.edu/sims/ph-scale-basics/1.5/

For example, this set of rules has a rule for every phetioID that contains "pHMeterNode":

```
    // Rename pHMeterNode => pHAccordionBox
    new TandemFragmentRenamed( 'phScale.microScreen.view.pHMeterNode.numberDisplay.valueText.stringProperty',
      'phScale.microScreen.view.pHAccordionBox.numberDisplay.valueText.stringProperty' ),
    new TandemFragmentRenamed( 'phScale.microScreen.view.pHMeterNode.numberDisplay.visibleProperty',
      'phScale.microScreen.view.pHAccordionBox.numberDisplay.visibleProperty' ),
    new TandemFragmentRenamed( 'phScale.microScreen.view.pHMeterNode.expandedProperty',
      'phScale.microScreen.view.pHAccordionBox.expandedProperty' ),
    new TandemFragmentRenamed( 'phScale.microScreen.view.pHMeterNode.expandCollapseButton.enabledProperty',
      'phScale.microScreen.view.pHAccordionBox.expandCollapseButton.enabledProperty' ),
    new TandemFragmentRenamed( 'phScale.microScreen.view.pHMeterNode.expandCollapseButton.visibleProperty',
      'phScale.microScreen.view.pHAccordionBox.expandCollapseButton.visibleProperty' ),
    new TandemFragmentRenamed( 'phScale.microScreen.view.pHMeterNode.expandCollapseButton.property',
      'phScale.microScreen.view.pHAccordionBox.expandCollapseButton.property' ),
    new TandemFragmentRenamed( 'phScale.microScreen.view.pHMeterNode.visibleProperty',
      'phScale.microScreen.view.pHAccordionBox.visibleProperty' ),
    new TandemFragmentRenamed( 'phScale.microScreen.view.pHMeterNode.pHProperty',
      'phScale.microScreen.view.pHAccordionBox.pHProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.spinner.numberDisplay.valueText.stringProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.spinner.numberDisplay.valueText.stringProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.spinner.numberDisplay.visibleProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.spinner.numberDisplay.visibleProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.spinner.incrementButton.enabledProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.spinner.incrementButton.enabledProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.spinner.incrementButton.visibleProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.spinner.incrementButton.visibleProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.spinner.decrementButton.enabledProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.spinner.decrementButton.enabledProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.spinner.decrementButton.visibleProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.spinner.decrementButton.visibleProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.spinner.visibleProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.spinner.visibleProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.spinner.enabledProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.spinner.enabledProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.spinner.property',
      'phScale.mySolutionScreen.view.pHAccordionBox.spinner.property' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.expandedProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.expandedProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.expandCollapseButton.enabledProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.expandCollapseButton.enabledProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.expandCollapseButton.visibleProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.expandCollapseButton.visibleProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.expandCollapseButton.property',
      'phScale.mySolutionScreen.view.pHAccordionBox.expandCollapseButton.property' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.pHMeterNode.visibleProperty',
      'phScale.mySolutionScreen.view.pHAccordionBox.visibleProperty' ),
```

Can be safely collapsed to this (note that `macroScreen.view.pHMeterNode` still exists):

```
// Rename pHMeterNode => pHAccordionBox
new TandemFragmentRenamed( 'microScreen.view.pHMeterNode', 'microScreen.view.pHAccordionBox' ),
new TandemFragmentRenamed( 'mySolutionScreen.view.pHMeterNode', 'mySolutionScreen.view.pHAccordionBox' ),
```

**Comment by pixelzoom** [comment]
In the above commits, I consolidated many of the rules, and added GitHub issue links were possible.  This feels much more readable to me. 

Two things that I'd like @samreid or @zepumph to review:

(1) Review the consolidation, let me know if you think this is a good pattern/direction.

(2) Investigate this possible bug in migration processing:

The set of rules for "Rename concentrationLabel => concentrationText" contained these 2 rules:

```
    new TandemFragmentRenamed( 'phScale.microScreen.view.graphNode.graphControlPanel.graphUnitsSwitch.concentrationLabel.stringProperty',
      'phScale.microScreen.view.graphNode.graphControlPanel.graphUnitsSwitch.concentrationText.stringProperty' ),
    new TandemFragmentRenamed( 'phScale.mySolutionScreen.view.graphNode.graphControlPanel.graphUnitsSwitch.concentrationLabel.stringProperty',
      'phScale.mySolutionScreen.view.graphNode.graphControlPanel.graphUnitsSwitch.concentrationText.stringProperty' ),
``

But `concentrationText.stringProperty` is now a DerivedProperty, so should have been deleted, similar to `quantityLabel.stringProperty`:

```
   // These are now composed strings based on dependencies, so are difficult to support migration. Punt.
    new TandemFragmentDelete( 'phScale.microScreen.view.graphNode.graphControlPanel.graphUnitsSwitch.quantityLabel.stringProperty' ),
    new TandemFragmentDelete( 'phScale.mySolutionScreen.view.graphNode.graphControlPanel.graphUnitsSwitch.quantityLabel.stringProperty' ),
```

I ended up fixing and consoldating like this:

```
    // These are now DerivedProperties, so are difficult to support migration. Punt.
    // NOTE! These must be done before rename of quantityLabel and concentrationLabel below.
    new TandemFragmentDelete( 'graphUnitsSwitch.quantityLabel.stringProperty' ),
    new TandemFragmentDelete( 'graphUnitsSwitch.concentrationLabel.stringProperty' ),
```

I'm not sure how/why this was passing migration, but it might represent a error somewhere in migration processing.

**Comment by pixelzoom** [comment]
Oops, I need to do the same for ph-scale-basics. Self-assigning.

**Comment by pixelzoom** [comment]
ph-scale-basics-migration-rules.ts has been similarly revised.

**Comment by samreid** [comment]
This looks great, thanks!  My main question is about whether we should be using substring exact matches vs regular expressions.  What's there seems production-ready, but would be good to discuss substring exact matches vs regular expressions.

**Comment by pixelzoom** [comment]
12/22/22 phet-io meeting

@samreid will review.

**Comment by pixelzoom** [comment]
> My main question is about whether we should be using substring exact matches vs regular expressions.

12/22/2022 phet-io meeting: This will not be addressed for 1.6 release.  We could consolidate using regexs in future releases. Closing.
